### ISSUE TITLE
Reordering first commit corrupts commit

### ISSUE THREAD

**Issue by iitalics** [magit/magit#3574]
Hello. I recently encountered what I think is a minor bug with rebasing. If I create an empty git repository, issue two commits, then reorder them with interactive rebase, one of them goes away and a strange commit is left behind.

Initial state:
```
milo$ git init; ls -l
Initialized empty Git repository in /home/milo/magittesting/.git/
total 0
-rw-r--r-- 1 milo milo 0 Sep  1 14:13 a
-rw-r--r-- 1 milo milo 0 Sep  1 14:13 b
```

```
In the beginning there was darkness

Untracked files (2)
a
b
```

Make two commits:

```
Head:     master commit b

Recent commits
20c7953 master commit b
ec35104 commit a
```

Interactive rebase starting at `commit a` (using `r` `i`), swapping their order:

```
pick 20c7953 commit b
pick ec35104 commit a
```

Finish with `C-c C-c`. Then magit status shows:

```
Head:     master commit a

Recent commits
caa87cc master commit a
        68f249f\0\0Milo Turner\0\0commit b    
```

Recently fetched the latest version of magit. `M-x magit-status`:
**Magit 2.13.0-203-g854fb0b0, Git 2.18.0, Emacs 26.1, gnu/linux**

I do not think this has anything to do with my other installed packages. Magit does not crash or anything of that sort. Maybe this is an issue with git rather than magit, but I'm not very experienced with rebasing from the command line. Thanks.

**Comment by kyleam** [comment]
Thanks for reporting.  I can reproduce.  I haven't dug deeper, but note that it doesn't seem to be an issue with the rebase.  If you look at the history outside of magit (e.g., `git log -p`), you should see that everything looks as expected after the rebase.  Instead it looks to be a display issue.

**Comment by kyleam** [comment]
Hmm it seems like the issue is that in this situation the new root loses its author timestamp:

```
$ git log --format="%h %at"
94606c7 1535846374
6632064
```

And that seems to be a bug in Git:

```
% git cat-file -p 6632064
tree 6be660545b31f61a82a87d2b1915f0b88bb9f16f
author Kyle Meyer <kyle@kyleam.com> @1535846384 -04000yle@kyleam.com
committer Kyle Meyer <kyle@kyleam.com> 1535846396 -0400

b
```

Note the mangling of the author date field.

This is with Git v2.18.0, and the above history was created directly from the command line.

**Comment by kyleam** [comment]
Looks like this is already fixed in what will be Git 2.19.0.  See ca3e1826a0 (sequencer: fix "rebase -i --root" corrupting author header), 0f16c09aae (sequencer: fix "rebase -i --root" corrupting author header timezone), and 67f16e3d3f (sequencer: fix "rebase -i --root" corrupting author header timestamp).
